src/components/CellsContainer/CellsContainer.svelte

```diff
@@ -0,0 +1,201 @@
+  export let gap = 1;
+  export let getAdditionalRectProps;
+  export let tooltipBackground = '#4a5568';
+  export let tooltipColor = 'MintCream';
+  export let color = 'SteelBlue';
+  export let hoverColor = 'IndianRed';
+  export let margin = { top: 20, left: 20, right: 20, bottom: 20 };
+
+  let containerRef = null;
+  let svg = null;
+  let g = null;
+  let data = [...Array(count)];
+  let containerWidth = 0;
+  let height = 0;
+  let innerWidth = 0;
+
+  $: {
+    if (containerWidth) {
+      innerWidth = containerWidth - margin.left - margin.right;
+
+      const rectSize = size + gap * 2;
+      const colsAmount = Math.floor(innerWidth / rectSize);
+      const rowsAmount = Math.ceil(count / colsAmount);
+
+      height = rectSize * rowsAmount + margin.top + margin.bottom;
+
+      data = createRectsData({
+        containerWidth: innerWidth,
+        count,
+        size,
+        gap,
+        getAdditionalRectProps,
+        color,
+        hoverColor
+      });
+
+      updateSvgElementsSizes();
+      updateRects();
+    }
+  }
+
+  const keyGetter = (key) => (r) => r[key];
+
+  onMount(() => {
+    svg = createSvg(containerWidth, height);
+    g = svg.append('g');
+
+    containerRef.appendChild(svg.node());
+
+    g.selectAll('rect')
+      .data(data)
+      .enter()
+      .append('rect')
+      .attr('rx', 3)
+      .attr('ry', 3)
+      .style('cursor', 'pointer')
+      .on('mouseover', handleRectHover)
+      .on('mouseout', handleRectMouseout);
+  });
+
+  function updateSvgElementsSizes() {
+    svg && svg.attr('width', containerWidth).attr('height', height);
+    g && g.style('transform', `translate(${margin.left}px, ${margin.top}px)`);
+  }
+
+  function updateRects() {
+    if (g) {
+      const rects = g.selectAll('rect').data(data);
+      rects.exit().remove();
+      rects
+        .enter()
+        .append('rect')
+        .on('mouseover', handleRectHover)
+        .on('mouseout', handleRectMouseout)
+        .merge(rects)
+        .attr('fill', keyGetter('color'))
+        .attr('rx', 3)
+        .attr('ry', 3)
+        .attr('x', keyGetter('x'))
+        .attr('y', keyGetter('y'))
+        .attr('width', keyGetter('size'))
+        .attr('height', keyGetter('size'))
+        .style('cursor', 'pointer');
+    }
+  }
+
+  function handleRectHover(e, rect) {
+    select(this)
+      .transition()
+      .attr('fill', keyGetter('hoverColor'))
+      .duration(150);
+
+    if (rect.tooltipText) {
+      createTooltip(rect);
+    }
+  }
+
+  function createTooltip(rect) {
+    const xPadding = 5;
+    const yPadding = 2;
+    const textY = rect.y - 8;
+    const textX = rect.x + xPadding;
+
+    const [bgId, textId, triangleId] = [
+      getId('bg', rect.index),
+      getId('text', rect.index),
+      getId('triangle', rect.index)
+    ];
+
+    const bg = g.append('rect');
+
+    const text = g.append('text').text(rect.tooltipText);
+
+    const triangle = g.append('path');
+
+    setAttributes(text, {
```

Comment:
Well, it's a bit complex code structure.
At this point, I already lost is it still `onMount` or not...
Why `setAttributes` used 3 times?
I will understand that and will figure out what is happening, but for quick scanning, it's not very friendly )